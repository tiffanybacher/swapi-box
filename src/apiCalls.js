export const getRandomFilm = (ranNum) => {
  const url = `https://swapi.co/api/films/${ranNum}/`;

  return fetch(url)
    .then(response => {
      if (!response.ok) {
        throw Error('Film failed to load');
      } else {
        return response.json();
      }
    });
}