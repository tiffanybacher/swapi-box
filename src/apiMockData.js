  export default {

  "mockFilmResponse": {
    "title": "The Phantom Menace",
    "episode_id": 1,
    "opening_crawl": "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy Trade\r\nFederation has stopped all\r\nshipping to the small planet\r\nof Naboo.\r\n\r\nWhile the Congress of the\r\nRepublic endlessly debates\r\nthis alarming chain of events,\r\nthe Supreme Chancellor has\r\nsecretly dispatched two Jedi\r\nKnights, the guardians of\r\npeace and justice in the\r\ngalaxy, to settle the conflict....",
    "director": "George Lucas",
    "producer": "Rick McCallum",
    "release_date": "1999-05-19",
    "characters": [
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/11/",
        "https://swapi.co/api/people/16/",
        "https://swapi.co/api/people/20/",
        "https://swapi.co/api/people/21/",
        "https://swapi.co/api/people/32/",
        "https://swapi.co/api/people/33/",
        "https://swapi.co/api/people/34/",
        "https://swapi.co/api/people/36/",
        "https://swapi.co/api/people/37/",
        "https://swapi.co/api/people/38/",
        "https://swapi.co/api/people/39/",
        "https://swapi.co/api/people/40/",
        "https://swapi.co/api/people/41/",
        "https://swapi.co/api/people/42/",
        "https://swapi.co/api/people/43/",
        "https://swapi.co/api/people/44/",
        "https://swapi.co/api/people/46/",
        "https://swapi.co/api/people/48/",
        "https://swapi.co/api/people/49/",
        "https://swapi.co/api/people/50/",
        "https://swapi.co/api/people/51/",
        "https://swapi.co/api/people/52/",
        "https://swapi.co/api/people/53/",
        "https://swapi.co/api/people/54/",
        "https://swapi.co/api/people/55/",
        "https://swapi.co/api/people/56/",
        "https://swapi.co/api/people/57/",
        "https://swapi.co/api/people/58/",
        "https://swapi.co/api/people/59/",
        "https://swapi.co/api/people/47/",
        "https://swapi.co/api/people/35/"
    ],
    "planets": [
        "https://swapi.co/api/planets/8/",
        "https://swapi.co/api/planets/9/",
        "https://swapi.co/api/planets/1/"
    ],
    "starships": [
        "https://swapi.co/api/starships/40/",
        "https://swapi.co/api/starships/41/",
        "https://swapi.co/api/starships/31/",
        "https://swapi.co/api/starships/32/",
        "https://swapi.co/api/starships/39/"
    ],
    "vehicles": [
        "https://swapi.co/api/vehicles/33/",
        "https://swapi.co/api/vehicles/34/",
        "https://swapi.co/api/vehicles/35/",
        "https://swapi.co/api/vehicles/36/",
        "https://swapi.co/api/vehicles/37/",
        "https://swapi.co/api/vehicles/38/",
        "https://swapi.co/api/vehicles/42/"
    ],
    "species": [
        "https://swapi.co/api/species/1/",
        "https://swapi.co/api/species/2/",
        "https://swapi.co/api/species/6/",
        "https://swapi.co/api/species/11/",
        "https://swapi.co/api/species/12/",
        "https://swapi.co/api/species/13/",
        "https://swapi.co/api/species/14/",
        "https://swapi.co/api/species/15/",
        "https://swapi.co/api/species/16/",
        "https://swapi.co/api/species/17/",
        "https://swapi.co/api/species/18/",
        "https://swapi.co/api/species/19/",
        "https://swapi.co/api/species/20/",
        "https://swapi.co/api/species/21/",
        "https://swapi.co/api/species/22/",
        "https://swapi.co/api/species/23/",
        "https://swapi.co/api/species/24/",
        "https://swapi.co/api/species/25/",
        "https://swapi.co/api/species/26/",
        "https://swapi.co/api/species/27/"
    ],
    "created": "2014-12-19T16:52:55.740000Z",
    "edited": "2015-04-11T09:45:18.689301Z",
    "url": "https://swapi.co/api/films/4/"
  },

  "mockPeopleResponse": {
    "count": 87,
    "next": "https://swapi.co/api/people/?page=2",
    "previous": null,
    "results": [
        {
            "name": "Luke Skywalker",
            "height": "172",
            "mass": "77",
            "hair_color": "blond",
            "skin_color": "fair",
            "eye_color": "blue",
            "birth_year": "19BBY",
            "gender": "male",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/",
                "https://swapi.co/api/films/7/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [
                "https://swapi.co/api/vehicles/14/",
                "https://swapi.co/api/vehicles/30/"
            ],
            "starships": [
                "https://swapi.co/api/starships/12/",
                "https://swapi.co/api/starships/22/"
            ],
            "created": "2014-12-09T13:50:51.644000Z",
            "edited": "2014-12-20T21:17:56.891000Z",
            "url": "https://swapi.co/api/people/1/"
        },
        {
            "name": "C-3PO",
            "height": "167",
            "mass": "75",
            "hair_color": "n/a",
            "skin_color": "gold",
            "eye_color": "yellow",
            "birth_year": "112BBY",
            "gender": "n/a",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/5/",
                "https://swapi.co/api/films/4/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/2/"
            ],
            "vehicles": [],
            "starships": [],
            "created": "2014-12-10T15:10:51.357000Z",
            "edited": "2014-12-20T21:17:50.309000Z",
            "url": "https://swapi.co/api/people/2/"
        },
        {
            "name": "R2-D2",
            "height": "96",
            "mass": "32",
            "hair_color": "n/a",
            "skin_color": "white, blue",
            "eye_color": "red",
            "birth_year": "33BBY",
            "gender": "n/a",
            "homeworld": "https://swapi.co/api/planets/8/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/5/",
                "https://swapi.co/api/films/4/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/",
                "https://swapi.co/api/films/7/"
            ],
            "species": [
                "https://swapi.co/api/species/2/"
            ],
            "vehicles": [],
            "starships": [],
            "created": "2014-12-10T15:11:50.376000Z",
            "edited": "2014-12-20T21:17:50.311000Z",
            "url": "https://swapi.co/api/people/3/"
        },
        {
            "name": "Darth Vader",
            "height": "202",
            "mass": "136",
            "hair_color": "none",
            "skin_color": "white",
            "eye_color": "yellow",
            "birth_year": "41.9BBY",
            "gender": "male",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [],
            "starships": [
                "https://swapi.co/api/starships/13/"
            ],
            "created": "2014-12-10T15:18:20.704000Z",
            "edited": "2014-12-20T21:17:50.313000Z",
            "url": "https://swapi.co/api/people/4/"
        },
        {
            "name": "Leia Organa",
            "height": "150",
            "mass": "49",
            "hair_color": "brown",
            "skin_color": "light",
            "eye_color": "brown",
            "birth_year": "19BBY",
            "gender": "female",
            "homeworld": "https://swapi.co/api/planets/2/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/",
                "https://swapi.co/api/films/7/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [
                "https://swapi.co/api/vehicles/30/"
            ],
            "starships": [],
            "created": "2014-12-10T15:20:09.791000Z",
            "edited": "2014-12-20T21:17:50.315000Z",
            "url": "https://swapi.co/api/people/5/"
        },
        {
            "name": "Owen Lars",
            "height": "178",
            "mass": "120",
            "hair_color": "brown, grey",
            "skin_color": "light",
            "eye_color": "blue",
            "birth_year": "52BBY",
            "gender": "male",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/5/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [],
            "starships": [],
            "created": "2014-12-10T15:52:14.024000Z",
            "edited": "2014-12-20T21:17:50.317000Z",
            "url": "https://swapi.co/api/people/6/"
        },
        {
            "name": "Beru Whitesun lars",
            "height": "165",
            "mass": "75",
            "hair_color": "brown",
            "skin_color": "light",
            "eye_color": "blue",
            "birth_year": "47BBY",
            "gender": "female",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/5/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [],
            "starships": [],
            "created": "2014-12-10T15:53:41.121000Z",
            "edited": "2014-12-20T21:17:50.319000Z",
            "url": "https://swapi.co/api/people/7/"
        },
        {
            "name": "R5-D4",
            "height": "97",
            "mass": "32",
            "hair_color": "n/a",
            "skin_color": "white, red",
            "eye_color": "red",
            "birth_year": "unknown",
            "gender": "n/a",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/2/"
            ],
            "vehicles": [],
            "starships": [],
            "created": "2014-12-10T15:57:50.959000Z",
            "edited": "2014-12-20T21:17:50.321000Z",
            "url": "https://swapi.co/api/people/8/"
        },
        {
            "name": "Biggs Darklighter",
            "height": "183",
            "mass": "84",
            "hair_color": "black",
            "skin_color": "light",
            "eye_color": "brown",
            "birth_year": "24BBY",
            "gender": "male",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [],
            "starships": [
                "https://swapi.co/api/starships/12/"
            ],
            "created": "2014-12-10T15:59:50.509000Z",
            "edited": "2014-12-20T21:17:50.323000Z",
            "url": "https://swapi.co/api/people/9/"
        },
        {
            "name": "Obi-Wan Kenobi",
            "height": "182",
            "mass": "77",
            "hair_color": "auburn, white",
            "skin_color": "fair",
            "eye_color": "blue-gray",
            "birth_year": "57BBY",
            "gender": "male",
            "homeworld": "https://swapi.co/api/planets/20/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/5/",
                "https://swapi.co/api/films/4/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [
                "https://swapi.co/api/vehicles/38/"
            ],
            "starships": [
                "https://swapi.co/api/starships/48/",
                "https://swapi.co/api/starships/59/",
                "https://swapi.co/api/starships/64/",
                "https://swapi.co/api/starships/65/",
                "https://swapi.co/api/starships/74/"
            ],
            "created": "2014-12-10T16:16:29.192000Z",
            "edited": "2014-12-20T21:17:50.325000Z",
            "url": "https://swapi.co/api/people/10/"
        }
    ]
},

"mockPersonResponse": {
            "name": "Luke Skywalker",
            "height": "172",
            "mass": "77",
            "hair_color": "blond",
            "skin_color": "fair",
            "eye_color": "blue",
            "birth_year": "19BBY",
            "gender": "male",
            "homeworld": "https://swapi.co/api/planets/1/",
            "films": [
                "https://swapi.co/api/films/2/",
                "https://swapi.co/api/films/6/",
                "https://swapi.co/api/films/3/",
                "https://swapi.co/api/films/1/",
                "https://swapi.co/api/films/7/"
            ],
            "species": [
                "https://swapi.co/api/species/1/"
            ],
            "vehicles": [
                "https://swapi.co/api/vehicles/14/",
                "https://swapi.co/api/vehicles/30/"
            ],
            "starships": [
                "https://swapi.co/api/starships/12/",
                "https://swapi.co/api/starships/22/"
            ],
            "created": "2014-12-09T13:50:51.644000Z",
            "edited": "2014-12-20T21:17:56.891000Z",
            "url": "https://swapi.co/api/people/1/"
        },

"mockSpeciesResponse": {
    "name": "Human",
    "classification": "mammal",
    "designation": "sentient",
    "average_height": "180",
    "skin_colors": "caucasian, black, asian, hispanic",
    "hair_colors": "blonde, brown, black, red",
    "eye_colors": "brown, blue, green, hazel, grey, amber",
    "average_lifespan": "120",
    "homeworld": "https://swapi.co/api/planets/9/",
    "language": "Galactic Basic",
    "people": [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/4/",
        "https://swapi.co/api/people/5/",
        "https://swapi.co/api/people/6/",
        "https://swapi.co/api/people/7/",
        "https://swapi.co/api/people/9/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/11/",
        "https://swapi.co/api/people/12/",
        "https://swapi.co/api/people/14/",
        "https://swapi.co/api/people/18/",
        "https://swapi.co/api/people/19/",
        "https://swapi.co/api/people/21/",
        "https://swapi.co/api/people/22/",
        "https://swapi.co/api/people/25/",
        "https://swapi.co/api/people/26/",
        "https://swapi.co/api/people/28/",
        "https://swapi.co/api/people/29/",
        "https://swapi.co/api/people/32/",
        "https://swapi.co/api/people/34/",
        "https://swapi.co/api/people/43/",
        "https://swapi.co/api/people/51/",
        "https://swapi.co/api/people/60/",
        "https://swapi.co/api/people/61/",
        "https://swapi.co/api/people/62/",
        "https://swapi.co/api/people/66/",
        "https://swapi.co/api/people/67/",
        "https://swapi.co/api/people/68/",
        "https://swapi.co/api/people/69/",
        "https://swapi.co/api/people/74/",
        "https://swapi.co/api/people/81/",
        "https://swapi.co/api/people/84/",
        "https://swapi.co/api/people/85/",
        "https://swapi.co/api/people/86/",
        "https://swapi.co/api/people/35/"
    ],
    "films": [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/7/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/"
    ],
    "created": "2014-12-10T13:52:11.567000Z",
    "edited": "2015-04-17T06:59:55.850671Z",
    "url": "https://swapi.co/api/species/1/"
}

}
