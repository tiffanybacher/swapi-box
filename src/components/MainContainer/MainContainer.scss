.MainContainer {
  // height: 100px;
  background-color: white;
}