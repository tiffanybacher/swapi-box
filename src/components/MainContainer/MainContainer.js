import React, { Component } from 'react';
import { getCardData } from '../../apiCalls';
import ButtonContainer from '../ButtonContainer/ButtonContainer';
import CardHolder from '../CardHolder/CardHolder';

class MainContainer extends Component {
  state = {
    category: '',
    data: '',
    CardHolderShown: false
  }

  setCardData = (category) => {
    return getCardData(category)
      .then(data => this.setState({ 
        category,
        data: data.results,
        CardHolderShown: true
      }))
      .catch(error => this.setState({
        errorStatus: `Failed to load ${category} data`
      }));
  }

  render() {
    if (this.state.CardHolderShown) {
      var cardHolder = 
        <CardHolder 
          category={this.state.category} 
          data={this.state.data} />
    }

    return (
      <main className="MainContainer">
        <ButtonContainer 
          setCardData={this.setCardData} />
        {cardHolder}
      </main>
    );
  }
}

export default MainContainer;