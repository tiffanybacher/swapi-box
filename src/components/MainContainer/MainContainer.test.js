import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from './MainContainer';
import { mockPeopleResponse } from '../../apiMockData';
import { getCardData } from '../../apiCalls';
jest.mock('../../apiCalls');

describe('MainContainer', () => {
  let wrapper;
  let mockCategory;

  beforeEach(() => {
    wrapper = shallow(<MainContainer />);
    mockCategory = 'people';
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
      category: '',
      data: '',
      CardHolderShown: false
    });
  });

  describe('setCardData', () => {
    beforeAll(() => {
      getCardData.mockImplementation(() => {
        return Promise.resolve(mockPeopleResponse)
      });
    });

    it.skip('should set state with the passed category', async () => {
      await wrapper.instance().setCardData(mockCategory);
      
      expect(wrapper.state('category')).toEqual(mockCategory);
    });

    it.skip('should set state with the response data', async () => {
      await wrapper.instance().setCardData(mockCategory);

      expect(wrapper.state('data')).toEqual(mockPeopleResponse);
    });

    it('should set state with an error status if fetch fails', async () => {
      await wrapper.instance().setCardData(mockCategory);

      expect(wrapper.state('errorStatus')).toEqual(`Failed to load ${mockCategory} data`);
    });
  });
});