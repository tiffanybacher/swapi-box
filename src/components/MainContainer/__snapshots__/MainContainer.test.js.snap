// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer should match the snapshot 1`] = `
<main
  className="MainContainer"
>
  <ButtonContainer
    setCardData={[Function]}
  />
</main>
`;
